<a href="https://colab.research.google.com/github/nathfcastro/PROJECT-TWO-/blob/main/Project_Two.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project Two Part One 
Student Name: Nathalie Castro 

Whether you chose 2 pre-approved datasets or 2 datasets from another source, or one of each:

Create a notebook where you have uploaded and shown the .head() of each of your data sets. For each of the proposed datasets, include the following information in text cells:

**First choice: dataset 1**
> 1. Source of data
>2. Brief description of data
>3. What is the target?
>4. What does one row represent? (A person?  A business?  An event? A product?)
>5. Is this a classification or regression problem?
>6. How many features does the data have?
>7. How many rows are in the dataset?
>8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

**Second choice: dataset 2**
>1. Source of data
>2. Brief description of data
>3. What is the target?
>4. What does one row represent? (A person?  A business?  An event? A product?)
>5. Is this a classification or regression problem?
>6. How many features does the data have?
>7. How many rows are in the dataset?
>8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

# Data Set One: Cirrhosis

In [12]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt

In [13]:
filename = '/content/cirrhosis.csv'
cirrhosis = pd.read_csv(filename)
cirrhosis.head()

ID  N_Days Status             Drug    Age Sex Ascites Hepatomegaly Spiders  \
0   1     400      D  D-penicillamine  21464   F       Y            Y       Y   
1   2    4500      C  D-penicillamine  20617   F       N            Y       Y   
2   3    1012      D  D-penicillamine  25594   M       N            N       N   
3   4    1925      D  D-penicillamine  19994   F       N            Y       Y   
4   5    1504     CL          Placebo  13918   F       N            Y       Y   

  Edema  Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos    SGOT  \
0     Y       14.5        261.0     2.60   156.0    1718.0  137.95   
1     N        1.1        302.0     4.14    54.0    7394.8  113.52   
2     S        1.4        176.0     3.48   210.0     516.0   96.10   
3     S        1.8        244.0     2.54    64.0    6121.8   60.63   
4     N        3.4        279.0     3.53   143.0     671.0  113.15   

   Tryglicerides  Platelets  Prothrombin  Stage  
0          172.0      190.0         12.2    4.0  
1           88.0      221.0         10.6    3.0  
2           55.0      151.0         12.0    4.0  
3           92.0      183.0         10.3    4.0  
4           72.0      136.0         10.9    3.0

In [14]:
cirrhosis.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 20 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   ID             418 non-null    int64  
 1   N_Days         418 non-null    int64  
 2   Status         418 non-null    object 
 3   Drug           312 non-null    object 
 4   Age            418 non-null    int64  
 5   Sex            418 non-null    object 
 6   Ascites        312 non-null    object 
 7   Hepatomegaly   312 non-null    object 
 8   Spiders        312 non-null    object 
 9   Edema          418 non-null    object 
 10  Bilirubin      418 non-null    float64
 11  Cholesterol    284 non-null    float64
 12  Albumin        418 non-null    float64
 13  Copper         310 non-null    float64
 14  Alk_Phos       312 non-null    float64
 15  SGOT           312 non-null    float64
 16  Tryglicerides  282 non-null    float64
 17  Platelets      407 non-null    float64
 18  Prothrombi

In [15]:
cirrhosis.shape

(418, 20)

## How many features does the data have?
* There are 19 features in this data set. 

## What is the target?
* The target is Drug (placebo or D-penicillamine)

## How many rows are in the dataset?

In [16]:
# Display the number of rows and columns for the dataframe
cirrhosis.shape
print(f'There are {cirrhosis.shape[0]} rows, and {cirrhosis.shape[1]} columns.')
print(f'The rows represent {cirrhosis.shape[0]} observations, and the columns represent {cirrhosis.shape[1]-1} features and 1 target variable.')

There are 418 rows, and 20 columns.
The rows represent 418 observations, and the columns represent 19 features and 1 target variable.


## Brief Decscription of Data 

Data Source: https://www.kaggle.com/datasets/fedesoriano/cirrhosis-prediction-dataset

Total of 424 PBC patients that was referred by ther Mayo clinic during a ten-year interval had met an eligible criteria for a randomized placebo-control trial of the drug D-penicillamine. For the first 312 patients that participated in this clinical trial. However, 112 patients did not participate in the clinical trial, but consented to have the basic measurements recorded and siz of the cases were lost to follow-up shortly after diagnosis. 

Below is a description of the columns: 

> * ID: Unique Identifier 
* N_Days: number of days between registration and the earlier of death, transplantation, or study analysis time in July 1986
* Status: status of the patient C (censored), CL (censored due to liver tx), or D (death)
* Drug: type of drug D-penicillamine or placebo
* Age: age in [days]
* Sex: M (male) or F (female)
* Ascites: presence of ascites N (No) or Y (Yes)
* Hepatomegaly: presence of hepatomegaly N (No) or Y (Yes)
* Spiders: presence of spiders N (No) or Y (Yes)
* Edema: presence of edema N (no edema and no diuretic therapy for edema), S (edema present without diuretics, or edema resolved by diuretics), or Y (edema despite diuretic therapy)
* Bilirubin: serum bilirubin in [mg/dl]
* Cholesterol: serum cholesterol in [mg/dl]
* Albumin: albumin in [gm/dl]
* Copper: urine copper in [ug/day]
* Alk_Phos: alkaline phosphatase in [U/liter]
* SGOT: SGOT in [U/ml]
* Triglycerides: triglicerides in [mg/dl]
* Platelets: platelets per cubic [ml/1000]
* Prothrombin: prothrombin time in seconds [s]
* Stage: histologic stage of disease (1, 2, 3, or 4)






## Is this a classification or regression problem?
> * This is a classification problem 

## What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
> * Some challanges I forsee is probably exploring this data because there are a lot of columns in this data set. Additionally, I may have some challanges with cleaning because there may be rows that I have to replace because the data is not the same or there maybe a lot of missing data that I have to impute. 

# Data Set: Stroke

In [30]:
filename2 = '/content/healthcare-dataset-stroke-data.csv'
stroke = pd.read_csv(filename2)
stroke.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [31]:
stroke.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


In [32]:
stroke.shape

(5110, 12)

In [33]:
stroke['heart_disease'].value_counts()

0    4834
1     276
Name: heart_disease, dtype: int64

## How many features does the data have?
* There are 11 features in this data set. 

## What is the target?
* The target is stroke (0 or 1)

## How many rows are in the dataset?

In [34]:
# Display the number of rows and columns for the dataframe
stroke.shape
print(f'There are {stroke.shape[0]} rows, and {stroke.shape[1]} columns.')
print(f'The rows represent {stroke.shape[0]} observations, and the columns represent {stroke.shape[1]-1} features and 1 target variable.')

There are 5110 rows, and 12 columns.
The rows represent 5110 observations, and the columns represent 11 features and 1 target variable.


## Brief Decscription of Data 
Data Source: https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset

"This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status. Each row in the data provides relavant information about the patient." (fedesoriano)

Below is a description of the columns: 
> * id: unique identifier
* gender: "Male", "Female" or "Other"
* age: age of the patient
* hypertension: 0 if the patient doesn't have hypertension, 1 if the patient has hypertension
* heart_disease: 0 if the patient doesn't have any heart diseases, 1 if the patient has a heart disease
* ever_married: "No" or "Yes"
* work_type: "children", "Govt_jov", "Never_worked", "Private" or "Self-employed"
* Residence_type: "Rural" or "Urban"
* avg_glucose_level: average glucose level in blood
* bmi: body mass index
* smoking_status: "formerly smoked", "never smoked", "smokes" or "Unknown"*
* stroke: 1 if the patient had a stroke or 0 if not

*Note: "Unknown" in smoking_status means that the information is unavailable for this patient

## Is this a classification or regression problem?
> * This is a classification problem 

## What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
> * Some challanges I forsee is modeling the data set. Again, there could be a number of missing data that I will encounter in the data set which I will have trouble imputing or could cause data leakage. 